azavea/raster-foundry
# Add initial UI for Area of Interest projects

## Overview

This PR adds an initial UI for Area of Interest projects.

The main `projects/{projectid}/edit` page has been adjusted to handle an AOI project (see demo below).

This adds two routes and corresponding templates:
- `projects/edit/{projectid}/aoi-parameters` - houses the page to adjust AOI parameters (filters, frequency, etc.)
- `projects/edit/{projectid}/aoi-approve` - houses the page to approve AOI scenes when an AOI project is set to require manual approval for scene addition

This PR also updates the style of several sidepanels to match new mockups.

Most of the front-end is non-functional.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

The `projects/{projectid}/edit` page for a standard project.
<img width="553" alt="screen shot 2017-05-10 at 5 13 23 pm" src="https://cloud.githubusercontent.com/assets/2442245/25922023/a28ad11a-35a5-11e7-831b-fa99b7ba37ac.png">


The `projects/{projectid}/edit` page for an AOI project. Notice the tag and link in the sidebar header.
<img width="583" alt="screen shot 2017-05-10 at 5 13 42 pm" src="https://cloud.githubusercontent.com/assets/2442245/25922024/a28b7e26-35a5-11e7-925e-f0a7107c7089.png">


Then we step through the scene management pages with an AOI project.
<img width="578" alt="screen shot 2017-05-10 at 5 14 47 pm" src="https://cloud.githubusercontent.com/assets/2442245/25922077/d5f9261e-35a5-11e7-82ca-ef195de65c5b.png">
<img width="626" alt="screen shot 2017-05-10 at 5 14 54 pm" src="https://cloud.githubusercontent.com/assets/2442245/25922078/d5fbf4e8-35a5-11e7-970a-277baa6e3343.png">


Click the 'Approve Scenes' button takes the user to the...'Approve Scenes' page (`projects/edit/{projectid}/aoi-approve`). When we have some backend stuff hooked up this will have a list of the scenes to approve. Each scene will be able to be approved or rejected, and when the user is done assessing each scene (or however many scenes they feel like), they can click 'Approve Scenes'.
<img width="576" alt="screen shot 2017-05-10 at 5 28 22 pm" src="https://cloud.githubusercontent.com/assets/2442245/25922241/65d394f4-35a6-11e7-9833-6d9da6ade74a.png">

From the project edit page, if the user clicks the 'Edit AOI Parameters', they are taken to the `projects/edit/{projectid}/aoi-parameters` route.
<img width="548" alt="screen shot 2017-05-10 at 5 32 12 pm" src="https://cloud.githubusercontent.com/assets/2442245/25922314/bb16efb0-35a6-11e7-8520-08ec1aca6c9e.png">

There is a filter built in here that is a new component (rf-aoi-filter-pane) since the filters we are displaying are different than our standard browse filters.
<img width="890" alt="screen shot 2017-05-10 at 5 33 49 pm" src="https://cloud.githubusercontent.com/assets/2442245/25922340/da661792-35a6-11e7-8acf-1ede2231fdd2.png">


### Notes

The scene approval banner element is not dynamic and is shown to demonstrate the UI. I added it to most pages having to do with adding imagery so that it is consistent. I also added the banner to the main AOI project edit page so that a user would be aware of the need to approve scenes without having to first go into the 'Add/Remove Scenes' area.

I adjusted some wording to keep things consistent.

Our icons have a bit of extra space around them so headers that contain the back arrow seem pushed to the right a bit too much for my own comfort. Not sure of the best way to handle this.

## Testing Instructions

 * Set a project to an AOI project using PSQL. `update scenes set is_aoi_project=true where id='<scene_id>';`
 * Click through the project edit pages with your AOI project and see that everything looks about like the mockups and buttons take you where you think they should take you.
 * Do the same for a standard project to make sure I didn't break anything.

Closes #1669 
Closes #1665
